src/wazuh_modules/wm_database.c

```diff
@@ -117,6 +119,8 @@ void* wm_database_main(wm_database *data) {
     // the agent addition and removal from the database will be held by authd
     // in the master.
 #ifndef LOCAL
+    // Wait for inventory_sync to subscribe to router before sending deletion messages
+    sleep(5);
```

Comment:
Isn't there any other mechanism for waiting? if it is not possible to wait for some fd with select for example, (that I think not, because the fd is created by the publisher) couldn't we do a retry logic waiting for 500ms, 1s, 2s, 4s, etc? 

Just a question, I didn't check the entire logic.

Reply:
Since these are different threads of the same process, I couldn't think of a better solution; 1 second should be enough, but to be safe I added 5 seconds, and since the router simply discards the messages without giving errors if no consumer is ready, this is the simplest solution without major changes on the router side.